### PR TITLE
Fully automated release process

### PR DESCRIPTION
**Pull request by hadim**
Checklist:

- [x] Was this PR discussed in a issue? It is recommended to first discuss a new feature and let the community know whether you are planning or have started working on it before opening a PR.
- [ ] Add tests to cover the fixed bug(s) or the new introduced feature(s) (if appropriate).
- [ ] Update the API documentation if a new function is added or an existing one is deleted.
- [x] Added a `news` entry.
  - _copy `news/TEMPLATE.rst` to `news/my-feature-or-branch.rst`) and edit it._

---

Basically backporting https://github.com/datamol-io/datamol/issues/180 here.

### CODE REVIEW

**Comment by hadim** [comment]
(the edit on the README is just auto formatting)

**Review by maclandrol — APPROVED** [review]

**Comment by maclandrol** [comment]
LGTM, I think we should consider adding an environment (https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment)  to require approval before the `release`  action can be executed.
